### ISSUE TITLE
Feature: Import File Structure / Mindmapping Filemanager

### ISSUE THREAD

**Issue by sudoquai** [phase1geo/Minder#231]
Can we have the ability to import a file structure or maybe even a mindmapping filemanager ?

Freemind has the ability to import a file structure - it's incredible useful when you are going to sort your home or like in my case tasks. My task managment is pretty simple: Simply use your favourite filemanager and add directories in your nextcloud / google drive as tasks, a subtask would be a subdirectory: you can add pictures, etc. It works flawless on every system.

I've thought it would be usefull to import such a file structure and generate a mindmap for a better overview.

Next step maybe a mindmap filemanager ? What do you think ?

**Comment by phase1geo** [comment]
Just so that I can better understand the file structure feature.  If I understand what you are saying, the user would essentially open a directory which Minder would then parse the folder and all of its subfolders/contents.  All found directories would be subnodes of the main directory node and all files would be leaf nodes in the tree, correct?  If a file was detected to be an image file, Minder should display the image in the mind map.  If a text file was detected, it would show the entire contents of the file in the node text?

Is this a correct understanding of the feature?  Is there anything that I got wrong?

I will probably have some follow up questions about this, such as:

- Would alias file (symbolic links) look like node links in the mind map?
- Is there anything that would populate the note text for a node?
- Possibly other questions that I can't think of right now...

This is an interesting concept to me.  I've never heard of or thought of doing tasks in this way.  I suppose that if Minder could display tasks in this manner, it could also potentially create this directory structure from a mindmap.  Would that feature be desirable?

As for the file manager, there is already an opened feature request for that type of feature, I believe.  I'm assuming you are referring to a sidebar which would display a list of files and allow you to open them, delete them, etc.

**Comment by sudoquai** [comment]
Yes - exactly like you've described @phase1geo. Also you've mentioned some interesting things i didn't think about it yet like exporting the mindmap to a file structure.

This was the original idea i've had, when i've managed my flat directory task managment. I've thought: why there is no filemanager which displays a directory structure as a mindmap instead the usual linear view.
Mindmaps have been originally developed by IBM as far as i know view the intention that humans are more capable to manage mindmaps not linear lists.

It would be logically to have a filemanager with its functions operating on a mindmap view. I assume it would increase our all productivity.

Nevertheless importing and exporting mindmaps based on the directory structure would be extremly helpful for me (and maybe other people).
